app/src/main/java/com/islaharper/dawnofandroid/presentation/common/OneTap.kt

```diff
@@ -0,0 +1,129 @@
+package com.islaharper.dawnofandroid.presentation.common
+
+import android.app.Activity
+import android.util.Log
+import androidx.activity.compose.ManagedActivityResultLauncher
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.IntentSenderRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import com.google.android.gms.auth.api.identity.BeginSignInRequest
+import com.google.android.gms.auth.api.identity.SignInClient
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
+import com.islaharper.dawnofandroid.util.Constants.CLIENT_ID
+
+@Composable
+fun StartActivityForResult(
+    oneTapClient: SignInClient,
```

Comment:
I thought we were going to use a ViewModel instead of using a dependency in the Composables?

Reply:
Apologies, I refactored but didn't remove the old implementation after testing. Removed.